lib/ansible/plugins/strategy/linear.py

```diff
@@ -90,8 +83,6 @@ def _get_next_task_lockstep(self, hosts, iterator):
             if cur_task._uuid == task._uuid:
                 iterator.set_state_for_host(host.name, state)
                 host_tasks.append((host, task))
-            else:
-                host_tasks.append((host, noop_task))
```

Comment:
won't this break host task count parity/lockstep?

Reply:
I don't think so. I figured that instead of executing `noop` meta tasks the host can just be ignored for the current host loop iteration, which would actually be no op. The host would be again included in the host/task loop once its task is the "current" one. According to my testing and CI this does not break the lockstep.